packages/buidler-core/src/internal/buidler-evm/provider/node.ts

```diff
@@ -1172,4 +1356,56 @@ export class BuidlerNode {
       await this._stateManager.setStateRoot(initialStateRoot);
     }
   }
+
+  private _removeLogs(blockNumber: number) {
+    this._filters.forEach(filter => {
+      if (filter.type !== Type.LOGS_SUBSCRIPTION) {
+        return;
+      }
+
+      for (let i = filter.logs.length - 1; i >= 0; i--) {
+        if (
+          new BN(toBuffer(filter.logs[i].blockNumber!)).toNumber() <=
+          blockNumber
+        ) {
+          break;
+        }
+
+        filter.logs[i].removed = true;
```

Comment:
Ohhh, I see what you did here! Is this how ganache or other testing nodes work?

Reply:
It appears that ganache didn't implement removed logs, probably because of the revert functionality that can reorganize a large number of blocks. I suggest we remove this feature as well.